### ISSUE TITLE
Tx metadata support in CLI for creating transactions

### ISSUE THREAD

**Issue by rvl** [cardano-foundation/cardano-wallet#2076]
# Context

ADP-307

For completeness, the CLI should support metadata when creating transactions.

# Decision

Add a `--metadata` option to the CLI:

```
Usage: cardano-wallet transaction create [--port INT] WALLET_ID
                                         --payment PAYMENT
                                         [--metadata JSON]
  Create and submit a new transaction.

Available options:
  -h,--help                Show this help text
  --port INT               port used for serving the wallet API. (default: 8090)
  --payment PAYMENT        address to send to and amount to send separated by @,
                           e.g. '<amount>@<address>'
  --metadata JSON    extra application-specific data to include in the transaction. The argument must be encoded in JSON format, and it must match the metadata JSON schema.
```

I considered supporting loading a JSON file, either a `--metadata-file` option, or through `@` notation ala `curl --data-raw`. But it should be ok in all cases to simply put the entire JSON data as a command-line argument.

# Acceptance Criteria

1. It must be possible to supply metadata to `cardano-wallet transaction create`.
2. Metadata must be validated as well-formed json before POSTing the request. It may also be validated according to the metadata schema before sending to the wallet server.
3. Metadata format errors must be reported to the user.
4. Wiki/docs updated with information on how to set transaction metadata.

---

# Development 

- Implemented in PR ##2125.

# QA 

1. CLI golden tests updated in `lib/cli/test/unit/Cardano/CLISpec.hs`.
2. Unit tests for metadata command line option added to `lib/cli/test/unit/Cardano/CLISpec.hs` - "Tx Metadata JSON option".
3. TRANSMETA_CREATE_01 integration test added.

**Comment by KtorZ** [comment]
To be tackle at the very end I think or could even be moved in another story.

**Comment by piotr-iohk** [comment]
lgtm.

There is actually one issue that is related -> https://github.com/input-output-hk/cardano-wallet/issues/2169.

Also the  way `--metadata $(<tx-meta.json)` of using metadata in cmdline didn't work for me in hash shell, whether the content in a file was single or double quoted. (That was suggested in https://github.com/input-output-hk/cardano-wallet/wiki/TxMetadata#cli)

I have updated https://github.com/input-output-hk/cardano-wallet/wiki/TxMetadata#cli on how to use metadata in CLI (which worked for me) for Linux/Mac (putting metadata argument in single quotes) and Windows (metadata argument in double quotes and double quotes inside metadata JSON escaped).

Updated [Wallet Cmdline interface](https://github.com/input-output-hk/cardano-wallet/wiki/Wallet-command-line-interface#transaction-create) with metadata exemplary usage.
